src/main/daml/Daml/Finance/Claims/Util/Builders.daml

```diff
@@ -60,6 +61,39 @@ createFixRatePaymentClaims dateToTime schedule periodicSchedule useAdjustedDates
         couponRate ownerReceives dayCountConvention notional cashInstrumentCid
     in prepareAndTagClaims dateToTime couponClaims "Fix rate payment"
 
+-- | Calculate the claims for callable bond: fix rate amount for each payment date and a redemption
+-- amount at the end (unless called by the issuer).
+createCallableFixRateBondClaims : (Date -> Time) -> Schedule -> PeriodicSchedule -> Bool
+  -> Decimal -> DayCountConventionEnum -> Decimal -> Deliverable -> TaggedClaim
+createCallableFixRateBondClaims dateToTime schedule periodicSchedule useAdjustedDatesForDcf
+  couponRate dayCountConvention notional cashInstrumentCid =
+    let
+      couponDatesAdjusted = map (.adjustedEndDate) schedule
+      couponAmounts = map (\p ->
+          couponRate *
+          (calcPeriodDcf dayCountConvention p useAdjustedDatesForDcf
+            periodicSchedule.terminationDate periodicSchedule.frequency)
+        ) schedule
+      coupons = map (\c -> give $ scale (Const c) $ one cashInstrumentCid) couponAmounts
+      principal = give $ scale (Const notional) $ one cashInstrumentCid
+
+      called = ("CALLED",) principal
+      tagNotCalled = "NOT CALLED"
+      finalCouponDate = last couponDatesAdjusted
+      notCalledFinal = (tagNotCalled,) $ when (at finalCouponDate) $ principal
+
+      combineTagClaim tag cpnDate coupon notCalledClaim =
+        (tag,) $ when (at cpnDate) $ and coupon $ or called notCalledClaim
+
+      tailClaims = foldl (\acc (date, cpn) -> combineTagClaim tagNotCalled date cpn acc)
+        notCalledFinal $ reverse $ tail $ zip couponDatesAdjusted coupons
```

Comment:
```suggestion
        notCalledFinal . reverse . tail $ zip couponDatesAdjusted coupons
```

Fyi, I think this can also be written using function composition